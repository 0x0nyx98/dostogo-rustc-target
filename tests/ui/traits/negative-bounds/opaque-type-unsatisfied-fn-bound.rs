//@ compile-flags: -Znext-solver

#![feature(negative_bounds, unboxed_closures)]

fn produce() -> impl !Fn<(u32,)> {}
//~^ ERROR expected a `Fn(u32)` closure, found `()`
//~| ERROR expected a `Fn(u32)` closure, found `()`
//~| ERROR the size for values of type `impl !Fn<(u32,)>` cannot be known at compilation time
//~| ERROR expected a `Fn(u32)` closure, found `()`

fn main() {}
