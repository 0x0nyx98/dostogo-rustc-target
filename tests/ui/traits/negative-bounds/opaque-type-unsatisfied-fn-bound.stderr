error[E0277]: expected a `Fn(u32)` closure, found `()`
  --> $DIR/opaque-type-unsatisfied-fn-bound.rs:5:17
   |
LL | fn produce() -> impl !Fn<(u32,)> {}
   |                 ^^^^^^^^^^^^^^^^ expected an `Fn(u32)` closure, found `()`
   |
   = help: the trait bound `(): !Fn(u32)` is not satisfied

error[E0277]: expected a `Fn(u32)` closure, found `()`
  --> $DIR/opaque-type-unsatisfied-fn-bound.rs:5:34
   |
LL | fn produce() -> impl !Fn<(u32,)> {}
   |                                  ^^ expected an `Fn(u32)` closure, found `()`
   |
   = help: the trait bound `(): !Fn(u32)` is not satisfied

error[E0277]: expected a `Fn(u32)` closure, found `()`
  --> $DIR/opaque-type-unsatisfied-fn-bound.rs:5:1
   |
LL | fn produce() -> impl !Fn<(u32,)> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected an `Fn(u32)` closure, found `()`
   |
   = help: the trait bound `(): !Fn(u32)` is not satisfied

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
